Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -115,7 +115,13 @@ public void handleOnBackPressed() {
 
     @VisibleForTesting
     void finishDropInWithError(Exception e) {
-        setResult(RESULT_FIRST_USER, new Intent().putExtra(DropInResult.EXTRA_ERROR, e));
+        Intent intent = new Intent();
+        intent.putExtra(DropInResult.EXTRA_ERROR, e);
+
+        DropInResult2 result = new DropInResult2();
+        result.setError(e);
+        intent.putExtra(DropInResult2.EXTRA_DROP_IN_RESULT_2, result);
```

Comment:
I decided to duplicate the logic on setting the result as an extra on the intent. This ensures that existing integrations can access the drop in result or error using the same extra keys as before, and in `DropInActivityContract` we can use the new key to access the `DropInResult2` to be returned to the merchants using the new integration.